unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/EmotesCustomization/EmotesCustomizationComponentView.cs

```diff
@@ -0,0 +1,242 @@
+        }
+
+        public void CloseEmoteInfoPanel() { emoteInfoPanel.SetActive(false); }
+
+        public EmoteCardComponentView GetEmoteCardById(string emoteId) { return GetAllEmoteCards().FirstOrDefault(x => x.model.id == emoteId); }
+
+        public void SetActive(bool isActive) { gameObject.SetActive(isActive); }
+
+        internal void ClickOnEmote(string emoteId, string emoteName, int slotNumber, bool isAssignedInSelectedSlot)
+        {
+            if (!isAssignedInSelectedSlot)
+                EquipEmote(emoteId, emoteName, slotNumber);
+            else
+                UnequipEmote(emoteId, selectedSlot);
+
+            emoteInfoPanel.SetActive(false);
+        }
+
+        internal void OnEmoteSelected(string emoteId) { selectedCard = GetEmoteCardById(emoteId); }
+
+        internal void ConfigureEmotesPool()
+        {
+            emoteCardsPool = PoolManager.i.GetPool(EMOTE_CARDS_POOL_NAME);
+            if (emoteCardsPool == null)
+            {
+                emoteCardsPool = PoolManager.i.AddPool(
+                    EMOTE_CARDS_POOL_NAME,
+                    GameObject.Instantiate(emoteCardPrefab).gameObject,
+                    maxPrewarmCount: EMOTE_CARDS_POOL_PREWARM,
+                    isPersistent: true);
+
+                emoteCardsPool.ForcePrewarm();
+            }
+        }
+
+        internal EmoteCardComponentView InstantiateAndConfigureEmoteCard(EmoteCardComponentModel emotesInfo)
+        {
+            EmoteCardComponentView emoteGO = emoteCardsPool.Get().gameObject.GetComponent<EmoteCardComponentView>();
+            emoteGO.Configure(emotesInfo);
+            emoteGO.onMainClick.RemoveAllListeners();
+            emoteGO.onMainClick.AddListener(() => ClickOnEmote(emoteGO.model.id, emoteGO.model.name, selectedSlot, emoteGO.model.isAssignedInSelectedSlot));
+            emoteGO.onInfoClick.RemoveAllListeners();
+            emoteGO.onInfoClick.AddListener(() => OpenEmoteInfoPanel(
+                emoteGO.model,
+                emoteGO.rarityMark.gameObject.activeSelf ? emoteGO.rarityMark.color : Color.grey, 
+                emoteGO.emoteInfoAnchor));
+            emoteGO.onEmoteSelected -= OnEmoteSelected;
+            emoteGO.onEmoteSelected += OnEmoteSelected;
+
+            return emoteGO;
+        }
+
+        internal void OnSlotSelected(int slotNumber, string emoteId)
+        {
+            List<EmoteCardComponentView> currentEmoteCards = GetAllEmoteCards();
+            foreach (var existingEmoteCard in currentEmoteCards)
+            {
+                existingEmoteCard.SetEmoteAsAssignedInSelectedSlot(existingEmoteCard.model.assignedSlot == slotNumber);
+            }
+
+            onSlotSelected?.Invoke(emoteId, slotNumber);
+        }
+
+        internal List<EmoteCardComponentView> GetAllEmoteCards()
+        {
+            return emotesGrid
+                .GetItems()
+                .Select(x => x as EmoteCardComponentView)
+                .ToList();
```

Comment:
In the future it would be great to just calculate these lists at the beginning of the application (unless they can change in runtime) to avoid fetching and filtering all this data every time we call `GetAllEmoteCards()` the same happens with other methods that return similar new lists, but maybe if those lists change in runtime it's inevitable

Reply:
Yeah, it is a bit difficult because those list are very dynamic... at least in this specific feature. But of course, for other features where we can avoid it, we should!